airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx

```diff
@@ -78,18 +42,35 @@ const useConfig = () => {
     }
   }, [configString]);
 
-  return { configString, configJson, setConfigString };
-};
+  // streams
+  const streamListRead = useListStreams({ manifest: jsonManifest, config: configJson }).data;
+  const streams = useMemo(() => {
+    return streamListRead?.streams ?? [];
+  }, [streamListRead]);
+  const firstStreamName = streams.length > 0 ? streams[0].name : undefined;
```

Comment:
:art: code style (JS will not throw any IndexOutOfBound exception, but just return you `undefined` if you try to access something outside the range).

```suggestion
  const firstStreamName = streams[0]?.name;
```

Reply:
Discussed over slack. Tim's suggestion results in undesired typing, because TS treats `streams[x]` as always returning an object, so that approach does not result in a `string | undefined` typing, which is desired here.